### PR TITLE
[WebGPU] Abstract interface needs to use WebCore types, but its in PAL

### PR DESCRIPTION
**Pull request by litherum**
#### 37aaa17f0b0b96081c9135b94ce88c7516027892
<pre>
[WebGPU] Abstract interface needs to use WebCore types, but its in PAL
<a href="https://bugs.webkit.org/show_bug.cgi?id=258666">https://bugs.webkit.org/show_bug.cgi?id=258666</a>
rdar://111504672

Reviewed by Mike Wyrzykowski and Kimmo Kinnunen.

For example, ExternalTextureDescriptor needs to know what VideoFrame is, but VideoFrame is not
in PAL. (Ideally, it would be, one day, because it&apos;s currently in WebCore/platform, but it&apos;s
not practical to move it right now.)

This patch moves the WebGPU abstract interface, and its implementation, from PAL to WebCore, so
it can use WebCore types like VideoFrame. If/when all the dependant types, like VideoFrame,
eventually make it into PAL, then we can move this WebGPU implementation back, but until that
happens, keeing WebGPU in WebCore makes development easier (and probably more correct, because
we won&apos;t have to figure out how to decompose all relevant objects into PAL/WTF types and/or do
downcasts to derived descriptor types).

[truncating commit message because GitHub says it&apos;s too long]

Canonical link: <a href="https://commits.webkit.org/265675@main">https://commits.webkit.org/265675@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/32875502eb8e97108fb77f2f85336a01cf36ea31

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/11611 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11812 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/12177 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/13254 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/11060 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/11631 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/14199 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11792 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/13939 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11775 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/12628 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/9848 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/13676 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/11709 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9920 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/10543 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/17678 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/10993 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/10697 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13875 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/11091 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/9140 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/10267 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2789 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/14547 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/10947 "Built successfully") | | |

### CODE REVIEW

**Review by mwyrzykowski — APPROVED** [review]

File: Source/WebKit/Shared/PALArgumentCoders.serialization.in [hunk 1]
```diff
@@ -20,8 +20,8 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-header: <pal/graphics/WebGPU/WebGPUTextureFormat.h>
-enum class PAL::WebGPU::TextureFormat : uint8_t {
+header: <WebCore/WebGPUTextureFormat.h>
```
Comment (kkinnunen-apple) [inline Source/WebKit/Shared/PALArgumentCoders.serialization.in]: maybe in subsequent patches this file is renamed..

File: Source/WebKit/Scripts/webkit/messages.py [hunk 2]
```diff
@@ -659,54 +659,54 @@ def headers_for_type(type):
...
+        'WebCore::WebGPU::StorageTextureAccess': ['<WebCore/WebGPUStorageTextureAccess.h>'],
+        'WebCore::WebGPU::StoreOp': ['<WebCore/WebGPUStoreOp.h>'],
+        'WebCore::WebGPU::TextureAspect': ['<WebCore/WebGPUTextureAspect.h>'],
+        'WebCore::WebGPU::TextureDimension': ['<WebCore/WebGPUTextureDimension.h>'],
+        'WebCore::WebGPU::TextureFormat': ['<WebCore/WebGPUTextureFormat.h>'],
+        'WebCore::WebGPU::TextureSampleType': ['<WebCore/WebGPUTextureSampleType.h>'],
+        'WebCore::WebGPU::TextureUsageFlags': ['<WebCore/WebGPUTextureUsage.h>'],
+        'WebCore::WebGPU::TextureViewDimension': ['<WebCore/WebGPUTextureViewDimension.h>'],
+        'WebCore::WebGPU::VertexFormat': ['<WebCore/WebGPUVertexFormat.h>'],
+        'WebCore::WebGPU::VertexStepMode': ['<WebCore/WebGPUVertexStepMode.h>'],
```
Comment (kkinnunen-apple) [inline Source/WebKit/Scripts/webkit/messages.py]: Maybe in subsequent patches these lines are sorted.

**Review by kkinnunen-apple — APPROVED** [review]

File: Source/WebKit/Shared/PALArgumentCoders.serialization.in (hunk 1 above)
Comment (litherum) [reply Source/WebKit/Shared/PALArgumentCoders.serialization.in]: Oh, good point, this shouldn't be in PALArgumentCoders any more.

**Comment by litherum** [comment]
Feedback: We should move the .idl files back to where they were before.

**Pull request merged**